src/components/shared/DivWithTooltip.js

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import Tooltip from './Tooltip';
+import type { CssPixels } from '../../types/units';
+
+type Props = {
+  tooltip: React.Node,
+  children?: React.Node,
+};
+
+type State = {|
+  isMouseOver: boolean,
+  mouseX: CssPixels,
+  mouseY: CssPixels,
+|};
+
+/**
+ * This component provides a way to automatically insert a tooltip when mousing over
+ * a div.
+ */
+export default class DivWithTooltip extends React.PureComponent<Props, State> {
```

Comment:
This could be made more abstract, but I felt that making this a div reduces the complexity of the implementation.